test/test_ez.py

```diff
@@ -0,0 +1,220 @@
+        shutil.rmtree(dump_dir_out)
+
+
+@pytest.mark.parametrize("task_name,task_class", TASK_CLASSES)
+def test_task(task_name, task_class):
+    task = ez.task.get_easy_task(task_name)
+    assert issubclass(task, task_class)
+
+
+@pytest.mark.parametrize("task_name,task_class", TASK_CLASSES)
+def test_task_with_dataset(task_name, task_class):
+    task = ez.task.get_easy_task_with_dataset(task_name)
+    assert issubclass(task, task_class)
+
+
+@pytest.mark.parametrize(
+    "tr_dump,val_dump,tr_ds,val_ds,tr_dl,val_dl,test_case",
+    [
+        ["not None string" if i == "1" else None for i in format(idx_case, "06b")]
+        + [format(idx_case, "06b")]
+        for idx_case in range(64)
+    ],
+)
+def test_check_argument(tr_dump, val_dump, tr_ds, val_ds, tr_dl, val_dl, test_case):
+    print(test_case)
+    try:
+        ez.trainer.check_argument(
+            train_dump_dir=tr_dump,
+            valid_dump_dir=val_dump,
+            train_dataset=tr_ds,
+            valid_dataset=val_ds,
+            train_dataloader=tr_dl,
+            valid_dataloader=val_dl,
+        )
+        assert test_case in ("110000", "001100", "000011")
+    except ValueError:
+        assert test_case not in ("110000", "001100", "000011")
+
+
+@pytest.mark.parametrize("task_name,task_class", TASK_CLASSES)
+def test_load_config(task_name, task_class):
+    with tempfile.TemporaryDirectory() as temp_dir:
+        # first create demo config
+        config_path = Path(temp_dir) / "config.yaml"
+        config_path.write_text("""task: {task_name}""")
+        default_config = task_class.get_default_config()
+        easy_config = ez.config.from_yaml(task_name, config_path)
+
+        for k in default_config.keys():
+            assert default_config[k] == easy_config[k]
+
+
+@pytest.mark.parametrize("task_name,task_class", TASK_CLASSES)
+def test_update_finetune_config(task_name, task_class):
+    with tempfile.TemporaryDirectory() as temp_dir:
+        # first create demo config
+        config_path = Path(temp_dir) / "config.yaml"
+        config_path.write_text("""use_lora: true""")
+        pretrain_config = task_class.get_default_config()
+        easy_config = ez.config.update_finetune_config(
+            task_name, pretrain_config, config_path
+        )
+
+        for k, v in easy_config.items():
+            if k != "use_lora":
+                assert v == pretrain_config[k]
+            else:
+                assert v
+
+
+def test_sentencepiece_preparation():
+    with tempfile.TemporaryDirectory() as temp_dir:
+        temp_dir = Path(temp_dir)
+        dataset = generate_random_dataset(5)
+        data_inputs = {
+            "speech": ["wav.scp", "sound"],
+            "text": ["text", "text"],
+        }
+        ez.data.create_dump_file(temp_dir / "dump", dataset, data_inputs)
+        ez.preprocess.prepare_sentences([temp_dir / "dump" / "text"], temp_dir / "spm")
+
+        # It seems training an sentencepiece in CI and pytest may cause error.
```

Comment:
Maybe remove this?

Reply:
I left this part to test the preprocessing function for sentencepiece.